include/faabric/util/snapshot.h

```diff
@@ -40,13 +40,13 @@ class SnapshotDiff
                  uint32_t offsetIn,
                  std::span<const uint8_t> dataIn);
 
-    SnapshotDataType getDataType() const;
+    SnapshotDataType getDataType() const { return dataType; }
 
-    SnapshotMergeOperation getOperation() const;
+    SnapshotMergeOperation getOperation() const { return operation; }
 
-    uint32_t getOffset() const;
+    uint32_t getOffset() const { return offset; }
 
-    std::span<const uint8_t> getData() const;
+    std::span<const uint8_t> getData() const { return data; }
```

Comment:
These (and the ones below, lines 121, 123, 162) have to be marked `inline` to avoid linker errors if this file is included from multiple cpp files in the same library - everything else looks good to me

Reply:
Interesting, I've never encountered this problem before; I don't think we've declared any other header-defined functions `inline` in the codebase. In addition, this file is included in both `snapshot/SnapshotServer.cpp` and `snapshot/SnapshotRegistry.cpp` (i.e. in the same CMake library, not sure if that's what you mean by library here). Can you be more specific about where this would this cause a problem?

Reply:
Oh nevermind, apparently any definitions inside the `class {}` body are implicitly inline, I didn't know this

Reply:
Ah interesting, good to know!